### DISCUSSION TITLE
Add preconnect to Automatic Webfont Optimization

### DISCUSSION THREAD

**Discussion by joe-bell** [vercel/next.js#24645] [Ideas]
### Describe the feature you'd like to request

Congrats on the new Automatic Webfont Optimization release! I'm the maintainer of the [`next-google-fonts`](https://github.com/joe-bell/next-google-fonts) which has aimed to provide a slightly better Google Fonts experience in the meantime.

I tried out the latest release, and had no problems whatsoever, but I did notice that Next.js **doesn't** add a `preconnect` for the actual font files; [an approach that Harry Roberts recommends](https://csswizardry.com/2020/05/the-fastest-google-fonts/#google-fonts-async-snippet).

```jsx
 <link rel="preconnect" href="https://fonts.gstatic.com" crossOrigin="anonymous" />
```
(See https://github.com/joe-bell/next-google-fonts/blob/481684dd35e4b8e10748f5a611a1231d2b86725d/src/index.tsx#L29-L35)

I've manually added the tag to Next.js projects myself in the meantime – you can see this in action on [joebell.co.uk](https://joebell.co.uk) – but it would be great to have it working out-of-the-box for those who aren't necessarily familiar with font-loading strategies 

---

I did clone the repo and try adding this feature myself, but I couldn't quite work out a way to apply this tag only once (e.g. if multiple Google fonts urls are included). I'm sure someone more familiar with the codebase would find this straightforward though!

### Describe the solution you'd like

Add the `preconnect` by default (with a `data-preconnect=<provider>` attribute for debugging) 

### Describe alternatives you've considered

Add to the docs that users may be interested in adding this `preconnect` tag themselves

**Comment by prateekbh** [comment]
I agree this should be added as a part of the automatic optimization

**Comment by housseindjirdeh** [comment]
I was intending to add a lint rule to the Next.js ESLint [plugin](https://www.npmjs.com/package/@next/eslint-plugin-next) to flag a warning if `preconnect` is missing, but automating it is also a good idea 👍

**Reply by prateekbh** [reply]
I like how the adoption is growing 😊

**Reply by housseindjirdeh** [reply]
All thanks to you for kicking things off :)

Rule added here: https://github.com/vercel/next.js/pull/24766. @janicklas-ralph Feel free to remove the rule once this issue is resolved

**Comment by janicklas-ralph** [comment]
Thanks! This is a WIP on our side. I will update this issue once we're done.

**Comment by DoctorDerek** [comment]
Awesome, thanks @janicklas-ralph !!

Hopefully as this feature matures the weird "drop out" depicted in https://dev.to/ekafyi/first-impressions-on-next-js-automatic-font-optimization-32a1 will also get fixed.

Thanks for your hard work!
